pkg/virt-controller/watch/replicaset.go

```diff
@@ -208,7 +229,33 @@ func (c *VMReplicaSet) execute(key string) error {
 	return err
 }
 
-func (c *VMReplicaSet) scale(rs *virtv1.VirtualMachineReplicaSet, vms []virtv1.VirtualMachine) error {
+// orphan removes the owner reference of all VMs which are owned by the controller instance.
+// Workaround for https://github.com/kubernetes/kubernetes/issues/56348 to make no-cascading deletes possible
+func (c *VMReplicaSet) orphan(cm *controller.VirtualMachineControllerRefManager, rs *virtv1.VirtualMachineReplicaSet, vms []*virtv1.VirtualMachine) error {
+
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(vms))
+	wg.Add(len(vms))
+
+	for _, vm := range vms {
+		go func(vm *virtv1.VirtualMachine) {
+			defer wg.Done()
+			err := cm.ReleaseVirtualMachine(vm)
+			if err != nil {
+				errChan <- err
+			}
+		}(vm)
+	}
+	wg.Wait()
+	select {
+	case err := <-errChan:
+		return err
```

Comment:
any reason to aggregate these errors?

Reply:
They are not aggregated. It just executes a bunch of async operations, and are only interested in reporting the first error which occurs (the others are very likely the same). Therefore, I create a channel which can take as many errors as can occure at max (to not block any gorouting), but then I only thake the first one.

Reply:
works for me